src/mono/wasm/debugger/BrowserDebugProxy/MonoProxy.cs

```diff
@@ -470,6 +486,26 @@ protected override async Task<bool> AcceptCommand(MessageId id, string method, J
             return false;
         }
 
+        private async Task<bool> OnSetVariableValue(MessageId id, int scopeId, string varName, JToken varValue, CancellationToken token)
+        {
+            ExecutionContext ctx = GetContext(id);
+            Frame scope = ctx.CallStack.FirstOrDefault(s => s.Id == scopeId);
+            if (scope == null)
+                return false;
+            var varIds = scope.Method.GetLiveVarsAt(scope.Location.CliLocation.Offset);
+            if (varIds == null)
+                return false;
+            var varToSetValue = varIds.FirstOrDefault(v => v.Name == varName);
+            if (varToSetValue == null)
+                return false;
+            Result res = await SendMonoCommand(id, MonoCommands.SetVariableValue(scopeId, varToSetValue.Index, varName, varValue["value"].Value<string>()), token);
+            if (res.Value["result"]?["value"].Value<int>() == 1)
```

Comment:
What is this checking? Should this use `res.IsOk` instead?

Reply:
Done.